InstrgamBot Update

OsamaSaw opened the pull request:
hello there, 
I was looking for a code to help me with something, and your code was relatively close to what I was trying to do. 
so I updated your code, took some parts from it, and added them to the code I am working on. 

I hope you like the new updates.

have a good day 
Osama

OsamaSaw committed:
Updated the deprecated function
Added a better way to handle the chromium drive
Updated all the Xpaths
Added more try blocks to prevent unnecessary crashes

OsamaSaw committed:
Added the header

Engineervinay commented:
Hey @OsamaSaw , thanks for PR, but can you list out changes that you performed on code?

OsamaSaw commented:
Oh Hello @Engineervinay 
what I did was 
I added a better way to handle the chromium drive by implementing the ChromeDriverManager, by doing this no need to download the chromium and declaring the path, the ChromeDriverManager will do all this as you can see in line 55

I updated the deprecated functions from selenium and all the Xpaths and selectors

I added more try blocks to prevent unnecessary crashes sometimes the "not now button" doesn't always show up and the page sometimes takes too much to load or just lags, so these blocks prevent such crashes.

I removed most of the Sleep functions and replace them with "DriverWait Until EC.presence_of_element_located" there is no need to wait 20 sec if the element is located after 3, of course, I added a fallback with is max of 20 sec. ✌

duraki commented:
Looks like a great addition. WDYT about merge @Engineervinay?
